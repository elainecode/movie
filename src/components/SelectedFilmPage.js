import React, {Component} from 'react';
import FilmDetails from './FilmDetails'
import FilmList from './FilmList'


class SelectedFilmPage extends Component {
    constructor() {
        super()
            this.state = {
            
            }   
    }

    render() {
        return (
            <div>
            <h4>Selected Film Page</h4>
            <FilmDetails
                id={this.props.match.params.id}
            />
            <FilmList></FilmList>
            </div>
        )
    }
}

 export default SelectedFilmPage