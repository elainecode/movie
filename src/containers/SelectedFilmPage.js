import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';
import {
  loadSelectedFilm,
  resetSelectedToDefaultState,
} from '../actions';
import { scrollToContentDivTop } from '../helpers';

import FilmDetails from '../components/FilmDetails';

class SelectedFilmPage extends Component {
  componentDidMount() {
    const {
      closePopOver,
      anchorEl,
      filmStepper,
      removeKeyDownEvent,
    } = this;
    window.addEventListener(
      'click',
      e => {
        scrollToContentDivTop(e, anchorEl), closePopOver(e);
      },
      false,
    );
    window.addEventListener('keydown', filmStepper, false);
  }

  componentDidUpdate(prevProps) {
    console.log(this.props.selectedFilm.id);
    const {
      match: {
        params: { id, query },
      },
    } = prevProps;
    const {
      selectedFilm,
      history,
      match: { path },
    } = this.props;
    if (selectedFilm.id && selectedFilm.id != id) {
      const newPath = path
        .replace(':query', query)
        .replace(':id', selectedFilm.id);
      history.push(newPath);
    }
  }

  componentWillUnmount() {
    const { closePopOver, anchorEl, filmStepper } = this;
    window.removeEventListener(
      'click',
      e => {
        scrollToContentDivTop(e, anchorEl), closePopOver(e);
      },
      false,
    );
    window.removeEventListener('keydown', filmStepper, false);
  }

  closePopOver = e => {
    e.preventDefault();
    if (e.target.getAttributeNames()[0] === 'aria-hidden') {
      this.props.resetSelectedToDefaultState();
    }
  };

  anchorEl = () => {
    return document.body.querySelector('.content');
  };

  filmStepper = e => {
    if (e.target.tagName === 'INPUT') {
      e.stopPropagation();
      return;
    }
    e.preventDefault();
    switch (e.key) {
      case 'ArrowLeft':
        this.arrowPress('left');
        break;
      case 'ArrowRight':
        this.arrowPress('right');
        break;
      default:
        e.preventDefault();
        break;
    }
  };

  arrowPress = direction => {
    const { selectedFilm, films, loadSelectedFilm } = this.props;
    const end = films.length - 1;
    const currIndex = films.findIndex(
      ({ id }) => id === selectedFilm.id,
    );
    if (currIndex != end) {
      const filmId = films[(direction === 'right') ? currIndex + 1 : currIndex - 1].id;
      loadSelectedFilm(nextFilmId);
    }
  };

  render() {
    const { films, genres, selectedFilm } = this.props;
    const { anchorEl } = this;

    return (
      <>
        <FilmDetails
          loading={loading}
          film={selectedFilm}
          genres={genres}
          anchorEl={anchorEl()}
        />
      </>
    );
  }
}

const mapStateToProps = ({ selectedFilm, loading }) => ({
  selectedFilm,
  loading,
});

export default connect(
  mapStateToProps,
  { loadSelectedFilm, resetSelectedToDefaultState },
)(SelectedFilmPage);
