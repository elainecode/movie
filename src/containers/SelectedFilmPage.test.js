import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import SelectedFilmPage from './SelectedFilmPage';

const mockStore = configureStore([]);

describe('<SelectedFilmPage/>', () => {
  let store;
  const loadSelectedFilm = jest.fn();
  const resetSelectedToDefaultState = jest.fn();
  const clickFilm = jest.fn();
  const films = [{}];
  const genres = {
    28: 'Action',
    12: 'Adventure',
    14: 'Fantasy',
    36: 'History',
  };
  beforeEach(() => {
    store = mockStore({
      selectedFilm: {
        title: '',
        poster_path: '',
        release_date: '19992345678',
        runtime: '',
        director: '',
        genres: [28, 87, 14],
        cast: ['name1', 'name2', 'name3'],
      },
      loadSelectedFilm,
      resetSelectedToDefaultState,
    });
  });
  test('should render with given state from Redux store', () => {
    const component = mount(
      <Router
        render={routerProps => (
          <Provider store={store}>
            <SelectedFilmPage
              films={films}
              genres={genres}
              clickFilm={clickFilm}
              {...routerProps}
            />
          </Provider>
        )}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();

    component.unmount();
  });
});
