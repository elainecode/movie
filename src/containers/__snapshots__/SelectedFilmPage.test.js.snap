// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedFilmPage/> should render with given state from Redux store 1`] = `
<BrowserRouter
  render={[Function]}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />
</BrowserRouter>
`;
